// python -m grpc_tools.protoc -I. --python_out=. --pyi_out=. --grpc_python_out=. ./components/proto/EchoChatBE.proto

syntax = "proto3";

service EchoChatBE {
  rpc WSValidateToken(TokenValue) returns (AccountinfoValue) {}
  rpc WSNewMessage(NewMessageValue) returns (NewMessageResult) {}
}

message TokenValue {
  string token = 1;
}

message AccountinfoValue {
  int32 id = 1;
  string name = 2;
  int32 identifier = 3;
}

message NewMessageValue {
  int32 accountinfo_id = 1;
  string accountinfo_name = 2;
  string group_id = 3;
  string content = 4;
  string type = 5;
}

message NewMessageResult {
  bool result = 1;
  optional string error = 2;
}